### ISSUE TITLE
[Feature]: Configurable pacing controls for slow/local LLM workloads

### ISSUE THREAD

**Issue by otherippo** [zeroclaw-labs/zeroclaw#2963]
### Summary

Add configurable pacing controls for slow/local LLM workloads (browser automation and long multi-step tasks)

### Problem statement

When using ZeroClaw with a local LLM (Ollama, llama.cpp, vLLM, etc.), multi-step tasks that involve sequential tool use — especially the browser tool's navigate → screenshot → analyze → act cycle — get aborted before they can complete. This happens for two reasons that compound each other:

The message timeout scaling cap (×4) is hardcoded and not proportional to the actual number of steps a task needs. A slow first turn can consume a disproportionate share of the budget, causing the overall task to time out even though each individual step was making progress.
The loop detector is purely count-based. It has no awareness of elapsed time or of tool-specific expected behavior. The browser workflow structurally resembles a loop (screenshot taken, screenshot taken again) even when genuine forward progress is happening. Currently, users must disable loop detection entirely to avoid false positives, which removes a safety feature that is genuinely useful in other contexts.

#### Why this matters
ZeroClaw explicitly targets low-cost local hardware with on-device LLMs as a first-class use case. The browser tool is one of the most compelling autonomous capabilities. But these two features are currently in tension: the timeout and loop detection subsystems were clearly calibrated for fast cloud providers and short tasks. Making the pacing layer configurable — rather than requiring users to disable safety features entirely — would make ZeroClaw significantly more viable for the local-LLM, long-horizon-task use case that is central to the project's positioning.

### Proposed solution

Three targeted additions to config.toml, all under a new [slow_llm] or [pacing] section (or integrated into existing sections):
1. Per-step timeout instead of only a per-message timeout
A step_timeout_secs value that sets the maximum time allowed for a single LLM inference turn, independent of the total message budget. This lets the overall task run for as long as needed while still catching a truly hung model response. Today, the only option is the message-level budget, which creates an implicit tradeoff between "allow long tasks" and "catch infinite hangs."
2. Time-gated loop detection
A loop_detection_min_elapsed_secs threshold: loop detection only activates after a task has been running for at least this many seconds. Tasks completing in under 30 seconds continue to get aggressive loop protection; long-running tasks (browser automation, multi-page research) get a grace period before the detector starts counting.
Optionally, a per-tool loop detection override: for example, loop_ignore_tools = ["browser_screenshot", "browser_navigate"] to exclude visually-repetitive-but-progressing tools from the identical-output detector.
3. Timeout scaling factor as a config value
The current multiplier ceiling of 4 in the formula message_timeout_secs × min(max_tool_iterations, 4) is hardcoded. Exposing this as message_timeout_scale_max (or similar) would allow users working with slow local models on long tasks to scale the budget proportionally without needing to inflate the base timeout to compensate.

### Non-goals / out of scope

Changing the default values of any existing timeout or loop detection parameters. This request only adds new optional config keys; existing behavior for cloud-provider users remains unchanged.

Also out of scope (but would still be nice to have): adaptive/automatic tuning of timeouts based on observed LLM speed

### Alternatives considered

Not really usable alternatives since they deactivate useful safety features: 
- disable loop detection entirely to avoid false positives
- tradeoff between "allow long tasks" and "catch infinite hangs"
- inflate the base timeout to compensate

### Acceptance criteria

- step_timeout_secs can be set in config and correctly aborts a single inference turn that exceeds it, without terminating the overall task loop
- loop_detection_min_elapsed_secs delays loop-detection counting until the threshold has passed, verifiable by running a browser task with a slow local LLM that would previously trigger a false-positive abort
- message_timeout_scale_max overrides the hardcoded 4 ceiling in the scaling formula, with the effective budget correctly reflecting the new value
- Per-tool loop detection exclusions (loop_ignore_tools) prevent named tools from contributing to identical-output or alternating-pattern counts
- No regression in existing timeout or loop detection behavior when the new keys are absent (full backwards compatibility)

### Architecture impact

Low to moderate. The timeout scaling formula and loop detection counters live in well-isolated areas of the codebase. The new config keys extend the existing ReliabilityConfig struct and the channel message processing context. No new traits or subsystems are required. 

### Risk and rollback

Risk is low. All new parameters are strictly opt-in with no effect when absent. The only regression risk is in the loop detection wiring change for per-tool exclusions, which could inadvertently suppress loop detection for unintended tools if the exclusion list is misconfigured by a user — mitigated by clear documentation and by keeping the default exclusion list empty. 

Rollback is straightforward: removing the new config keys restores prior behavior entirely, as no existing code paths are modified, only extended

### Breaking change?

No

### Data hygiene checks

- [x] I removed personal/sensitive data from examples, payloads, and logs.
- [x] I used neutral, project-focused wording and placeholders.

**Comment by j00001001** [comment]
This should not be labeled as enhancement but as a bug, since the implications of hardcoded timeouts are worse. When trying to use the Nextcloud Talk channel in combination with a self hosted lmstudio, it simply won't work. The webhook is called, zeroclaw sends a request to lmstudio. lmstudio starts processing the prompt, but zeroclaw disconnects after 4 or 5 seconds. Here are some lmstudio logs:

[2026-03-07 17:14:54][INFO][gpt-oss-safeguard-120b] Running Anthropic messages API on conversation with 2 messages.
[2026-03-07 17:14:54][WARN][gpt-oss-safeguard-120b] No valid custom reasoning fields found in model 'lmstudio-community/gpt-oss-safeguard-120b-GGUF/gpt-oss-safeguard-120b-MXFP4-00001-of-00002.gguf'. Reasoning setting 'off' cannot be converted to any custom KVs.
[2026-03-07 17:14:54][DEBUG] Prompt successfully formatted with Harmony.
[2026-03-07 17:14:54][DEBUG] Prompt successfully formatted with Harmony.
[2026-03-07 17:14:54][DEBUG] LlamaV4::predict slot selection: session_id=<empty> server-selected (LCP/LRU)
[2026-03-07 17:14:54][DEBUG] slot get_availabl: id  2 | task -1 | selected slot by LCP similarity, sim_best = 0.320 (> 0.100 thold), f_keep = 1.000
[2026-03-07 17:14:54][DEBUG] slot launch_slot_: id  2 | task -1 | sampler chain: logits -> penalties -> ?dry -> ?top-n-sigma -> top-k -> ?typical -> top-p -> min-p -> ?xtc -> temp-ext -> dist 
slot launch_slot_: id  2 | task 662 | processing task, is_child = 0
slot update_slots: id  2 | task 662 | new prompt, n_ctx_slot = 65536, n_keep = 6404, task.n_tokens = 6404
slot update_slots: id  2 | task 662 | n_tokens = 2048, memory_seq_rm [2048, end)
slot update_slots: id  2 | task 662 | prompt processing progress, n_tokens = 2560, batch.n_tokens = 512, progress = 0.399750
[2026-03-07 17:14:54][INFO][gpt-oss-safeguard-120b] Prompt processing progress: 0.0%
[2026-03-07 17:14:55][DEBUG] slot update_slots: id  2 | task 662 | n_tokens = 2560, memory_seq_rm [2560, end)
[2026-03-07 17:14:55][DEBUG] slot update_slots: id  2 | task 662 | prompt processing progress, n_tokens = 3072, batch.n_tokens = 512, progress = 0.479700
[2026-03-07 17:14:55][INFO][gpt-oss-safeguard-120b] Prompt processing progress: 11.8%
[2026-03-07 17:14:56][DEBUG] slot update_slots: id  2 | task 662 | n_tokens = 3072, memory_seq_rm [3072, end)
[2026-03-07 17:14:56][DEBUG] slot update_slots: id  2 | task 662 | prompt processing progress, n_tokens = 3584, batch.n_tokens = 512, progress = 0.559650
[2026-03-07 17:14:56][INFO][gpt-oss-safeguard-120b] Prompt processing progress: 23.5%
[2026-03-07 17:14:57][DEBUG] slot update_slots: id  2 | task 662 | n_tokens = 3584, memory_seq_rm [3584, end)
[2026-03-07 17:14:57][DEBUG] slot update_slots: id  2 | task 662 | prompt processing progress, n_tokens = 4096, batch.n_tokens = 512, progress = 0.639600
[2026-03-07 17:14:57][INFO][gpt-oss-safeguard-120b] Prompt processing progress: 35.3%
[2026-03-07 17:14:59][DEBUG] slot update_slots: id  2 | task 662 | n_tokens = 4096, memory_seq_rm [4096, end)
slot update_slots: id  2 | task 662 | prompt processing progress, n_tokens = 4608, batch.n_tokens = 512, progress = 0.719550
[2026-03-07 17:14:59][INFO][gpt-oss-safeguard-120b] Prompt processing progress: 47.0%
[2026-03-07 17:14:59][INFO][LM STUDIO SERVER] Client disconnected. Stopping generation... (If the model is busy processing the prompt, it will finish first.)
